analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -1456,28 +1447,51 @@ public void ForFalse(object arg)
         {
             var canBeNull = arg?.ToString();
             FailsWhenFalse(notImportant, canBeNull != null, notImportant);
-            canBeNull.ToString();    // Noncompliant FP
+            canBeNull.ToString();    // Compliant
         }
 
         public void ForFalseAny(object arg1, object arg2, bool condition)
         {
             var canBeNull1 = arg1?.ToString();
             var canBeNull2 = arg2?.ToString();
-            object isNull = null;
             FailsWhenFalseAny(canBeNull1 != null, condition);
-            canBeNull1.ToString();  // Noncompliant FP
-            if (condition)
-            {
-                isNull.ToString();  // Noncompliant
-            }
-            else
-            {
-                isNull.ToString();  // Noncompliant FP, Unreachable
-            }
+            canBeNull1.ToString();  // Compliant
             FailsWhenFalseAny(condition, canBeNull2 != null);
             canBeNull2.ToString();  // Noncompliant FP
         }
 
+        public void BoolSymbols_TrueTrue()
+        {
+            var isTrue = true;
+            object isNull = null;
+            FailsWhenTrue(notImportant, isTrue, notImportant);
```

Comment:
isTrue can be inlined. (+3 more time below)

Reply:
I wanted to demonstrate that it's a field. It will unlikely be hardcoded